packages/passwordless-auth-construct/src/custom-auth/custom_auth_service.ts

```diff
@@ -151,32 +157,35 @@ export class CustomAuthService {
     // If the user is not found or if the attribute requested for challenge
     // delivery is not verified, return a fake successful response to prevent
     // user enumeration
-    if (event.request.userNotFound || !isVerified) {
-      logger.info(
-        'User not found or user does not have a verified phone/email.'
+    const validUser =
+      !event.request.userNotFound && (isVerified || isFirstSignInAttempt);
+
+    // If the user is found and if the attribute requested for challenge
+    // delivery is verified or if the user was created via passwordless sign up (first sign in attempt)
+    if (validUser) {
+      return challengeService.createChallenge(
+        { deliveryMedium, attributeName },
+        destination,
+        event
```

Comment:
nit/suggestion: IMO this function is easier to reason about when the happy case is at the bottom. Consider inversing the conditional and placing the user not found logic inside the if branch.